vcs/src/main/java/org/openjdk/skara/vcs/Repository.java

```diff
@@ -191,6 +191,18 @@ default void updateSubmodule(Submodule s) throws IOException {
         updateSubmodule(s.path());
     }
 
+    void addNote(Hash hash,
+                 List<String> note,
```

Comment:
Just out of curiosity, is there a specific reason for using `List<String>` in the interface and then concatenating the list? Why not use `String` directly?

Reply:
I just think it is more clear for a user of the API how lines are handled. If the method accepts `String hash` then it is (to me) harder to know if the API supports a single line or multiple lines (and the caller is responsible for inserting `\n` characters). With `List<String>` it is to me more clear, although I could have named the parameter `lines` to make it obvious 😄